File: libsignal/service/src/main/java/org/whispersystems/signalservice/api/payments/CurrencyConversion.java
package org.whispersystems.signalservice.api.payments;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.Map;

public final class CurrencyConversion {
  @JsonProperty
  private String base;

  @JsonProperty
  private Map<String, Double> conversions;

  public String getBase() {
    return base;
  }

  public Map<String, Double> getConversions() {
    return conversions;
  }
}

File: lib/libsignal-service/src/main/java/org/whispersystems/signalservice/api/payments/CurrencyConversion.kt
/*
 * Copyright 2025 Signal Messenger, LLC
 * SPDX-License-Identifier: AGPL-3.0-only
 */

package org.whispersystems.signalservice.api.payments

data class CurrencyConversion(
  val base: String = "",
  val conversions: Map<String, Double> = emptyMap()
)
